MelonLoader/Lemons/LemonEnumerator.cs

```diff
@@ -1,16 +1,27 @@
-﻿using System.Collections;
+﻿using System;
+using System.Collections;
 using System.Collections.Generic;
 using System.Linq;
 
 namespace MelonLoader
 {
-    public class LemonEnumerator<T> : IEnumerator
+    public class LemonEnumerator<T> : IEnumerator<T>, IEnumerable<T>
     {
         private T[] LemonPatch;
         private int NextLemon = 0;
 
+        /// <summary>
+        /// Creates a new instance of <see cref="LemonEnumerator{T}"/> with a new copy of '<paramref name="lemons"/>'.
+        /// </summary>
         public LemonEnumerator(T[] lemons)
-            => LemonPatch = lemons.ToArray();
+        {
+            LemonPatch = new T[lemons.Length];
```

Comment:
Having just dug around in the runtime reference source code, this is functionally identical to lemons.ToArray(), as ToArray has special handling for when `source` is an array itself, which ends up calling System.Array.CopyTo, which in turn calls System.Array.Copy. Basically, revert this, as it's more code for no benefit.

Reply:
Oh alright

Reply:
https://github.com/SlidyDev/MelonLoader/commit/55272e76e91cdfa64f3548308c44728783107d55